Apply ISR to reference particle (optional). (#1167)
BLAST-ImpactX/impactx: docs/source/usage/parameters.rst

  The number of terms retained in the Taylor series for the functions :math:`g(\chi)` and :math:`h(\chi)` appearing in equations (25) and (41) describing quantum effects.

> * ``algo.isr_ref_part`` (``integer`, optional, default: ``0``)

Reviewer: Since `amrex::ParmParse` supports `bool`, I would use this here (see for instance: `lattice.reverse`).

```suggestion
* ``algo.isr_ref_part`` (``boolean`, optional, default: ``false``)
```

> * ``algo.isr_ref_part`` (``boolean`, optional, default: ``false``)

Reviewer: I am not 100% sure yet on the naming. Maybe `algo.isr_on_ref` would be more descriptive? To discuss :)

Author: I think ...`_ref` by itself is not descriptive enough, but ...`_ref_part` is similar to naming used elsewhere.  What about `algo.isr_on_ref_part`?

Reviewer: Sounds great!

> * ``algo.isr_ref_part`` (``boolean`, optional, default: ``false``)

Author: ```suggestion
* ``algo.isr_on_ref_part`` (``boolean`, optional, default: ``false``)
```

>   Flag specifying whether ISR is to be applied to the reference particle.  When ``algo.isr_ref_part = false``, the reference particle does not lose energy due to radiation, and the

Author: ```suggestion
  Flag specifying whether ISR is to be applied to the reference particle.  When ``algo.isr_on_ref_part = false``, the reference particle does not lose energy due to radiation, and the
```

  mean energy of the beam particles will decrease.  This option is natural if the lattice optics, magnet settings, etc. are chosen without accounting for radiative energy loss.

>   When ``algo.isr_ref_part = 1``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.

Reviewer: ```suggestion
  Flag specifying whether ISR is to be applied to the reference particle.  When ``algo.isr_ref_part = false``, the reference particle does not lose energy due to radiation, and the
  mean energy of the beam particles will decrease.  This option is natural if the lattice optics, magnet settings, etc. are chosen without accounting for radiative energy loss.
  When ``algo.isr_ref_part = true``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.
```

>   When ``algo.isr_ref_part = true``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.

Author: ```suggestion
  When ``algo.isr_on_ref_part = true``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.
```

>   When ``algo.isr_on_ref_part = true``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.

Reviewer: @cemitch99 for symmetry, do you want to add a similar sentence for `true`? :)
```suggestion

  When ``algo.isr_on_ref_part = true``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.  This option is natural if ...
```

>   When ``algo.isr_on_ref_part = true``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.

Author: ```suggestion
  When ``algo.isr_on_ref_part = true``, the reference particle does lose energy due to radiation, and little centroid evolution is expected in the beam particles.  This option is natural if the lattice optics, magnet settings, etc. are chosen to account for radiative energy loss.
```
